### PR TITLE
fix: payroll attendance error

### PR DESCRIPTION
**Pull request by AfshanKhan**
### Steps to replicate:

- Create Employee joining July 25

- Create attendance for employee from 25th july to 31st july

- Run Payroll with attendance for July

#### Shows error that "Attendance not marked for employee" even when attendance is marked from the date of joining.
image

Now it considers start date either the "date of joining" or "start date" mentioned on basis of whichever is greater, hench solving this error.

### CODE REVIEW

**Review by Anurag810 — CHANGES_REQUESTED** [review]
image
As we know that payroll entry is for multiple employees. So, users have to search for every employee on the list. Make the message more generic and try to list which employee is left to mark attendance.

File: erpnext/payroll/doctype/payroll_entry/payroll_entry.py [hunk 1]
```diff
@@ -344,32 +344,36 @@ def validate_employee_attendance(self):
...
 
-	def get_count_holidays_of_employee(self, employee):
+	def get_count_holidays_of_employee(self, employee, start_date):
 		holiday_list = get_holiday_list_for_employee(employee)
 		holidays = 0
 		if holiday_list:
 			days = frappe.db.sql("""select count(*) from tabHoliday where
 				parent=%s and holiday_date between %s and %s""", (holiday_list,
-				self.start_date, self.end_date))
+				start_date, self.end_date))
```
Comment (Anurag810) [inline erpnext/payroll/doctype/payroll_entry/payroll_entry.py:367]: Avoid Query try to use ```frappe.get_all``` or ```frappe.get_list```

File: erpnext/payroll/doctype/payroll_entry/payroll_entry.py [hunk 2]
```diff
@@ -344,32 +344,36 @@ def validate_employee_attendance(self):
...
 				holidays = days[0][0]
 		return holidays
 
-	def get_count_employee_attendance(self, employee):
+	def get_count_employee_attendance(self, employee, start_date):
 		marked_days = 0
 		attendances = frappe.db.sql("""select count(*) from tabAttendance where
 			employee=%s and docstatus=1 and attendance_date between %s and %s""",
-			(employee, self.start_date, self.end_date))
+			(employee, start_date, self.end_date))
```
Comment (Anurag810) [inline erpnext/payroll/doctype/payroll_entry/payroll_entry.py]: Avoid Query try to use ```frappe.get_all``` or ```frappe.get_list```

**Comment by AfshanKhan** [comment]
> image
> As we know that payroll entry is for multiple employees. So, users have to search for every employee on the list. Make the message more generic and try to list which employee is left to mark attendance.

Screenshot 2020-11-13 at 3 36 19 PM
@Anurag810  The list of employees is shown below. Also listing all employees will make that error message unnecessarily big.

**Review by Anurag810 — CHANGES_REQUESTED** [review]

**Comment by Anurag810** [comment]
> > image
> > As we know that payroll entry is for multiple employees. So, users have to search for every employee on the list. Make the message more generic and try to list which employee is left to mark attendance.
> 
> Screenshot 2020-11-13 at 3 36 19 PM
> @Anurag810 The list of employees is shown below. Also listing all employees will make that error message unnecessarily big.

yeah got this.

File: erpnext/payroll/doctype/payroll_entry/payroll_entry.py (hunk 1 above)
Comment (AfshanKhan) [reply erpnext/payroll/doctype/payroll_entry/payroll_entry.py:367]: The queries are just to get count, I don't think using frappe.get_all or frappe.get_list is required here.

**Review by Anurag810 — APPROVED** [review]

**Pull request merged**